src/TauBounds.cpp

```diff
@@ -15,41 +16,41 @@ void calculateTauBounds() {
 
   for (int gen = 0; gen < dataSetup.numLoci; gen++) {
     int rootNodeId = dataState.lociData[gen]->root;
-    calculateLociTauBounds(rootNodeId, gen);
+    calculateLocusTauBounds(rootNodeId, gen);
   }
 
   propagateBoundsDownPopTree();
 
   if (DEBUG_TAU_BOUNDS) runTauBoundsAssertions();
 }
 
-int calculateLociTauBounds(int nodeId, int gen) {
+int calculateLocusTauBounds(int nodeId, int gen) {
   LikelihoodNode *currentNode = getNode(nodeId, gen);
   int pop = getNodePop(nodeId, gen);
 
   if (isLeafPop(pop))
     return pop;
 
-  int leftLca = calculateLociTauBounds(currentNode->leftSon, gen);
-  int rightLca = calculateLociTauBounds(currentNode->rightSon, gen);
+  int leftLca = calculateLocusTauBounds(currentNode->leftSon, gen);
+  int rightLca = calculateLocusTauBounds(currentNode->rightSon, gen);
 
   int lca_pop = lca_pops[leftLca][rightLca];
 
-  tau_bounds[lca_pop] = fmin(tau_bounds[lca_pop], currentNode->age);
+  tau_ubounds[lca_pop] = fmin(tau_ubounds[lca_pop], currentNode->age);
 
   return lca_pop;
 }
 
 void propagateBoundsDownPopTree() {
   for (int pop = 0; pop < dataSetup.popTree->numPops; pop++) {
-    updateTauBoundsOfDescendants(pop, tau_bounds[pop]);
+    updateTauBoundsOfDescendants(pop, tau_ubounds[pop]);
```

Comment:
this seems inefficient to me. Why is it not enough to call this function once with the root population and let all the information propagate dow the population phylogeny?